core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/definitions/ldap/implementations/ReleseablePool.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.accesscontrol.blocks.definitions.ldap.implementations
+
+import cats.Monad
+import cats.implicits._
+import monix.execution.atomic.Atomic
+
+import scala.language.higherKinds
+
+final class ReleseablePool[M[_] : Monad, A,B](acquire: B => M[A])(release: A => M[Unit]) {
```

Comment:
this class looks almost like cat's Resource https://typelevel.org/cats-effect/datatypes/resource.html

Reply:
It was intended to look almost like resource, but it has two main differences.
1. It's less safe. By default you can acquire resource, and not release it
2. It has a pool, so you can release single resource, or whole pool of resources.

Reply:
maybe we can use cat's resource and add lacking behaviour?

Reply:
It could be done, but I can't see profit. We're not using this pull in monadic. Using `Resource` we'll use unsafe API. Using `Resource` proper way, and benefit it's safety will require some refactor.

Reply:
It could be done, but I can't see profit. We're not using this pull in monadic. Using `Resource` we'll use unsafe API. Using `Resource` proper way, and benefit it's safety will require some refactor.